tools/selection_tools.cpp

```diff
@@ -606,6 +606,39 @@ void SelectSubsetEdgesByDirection(
 	                       maxDeviationAngle, selectFlipped);
 }
 
+Edge* SelectEdgeByMinPerpendicularDistance(
+			Mesh* m,
+			const vector3& coord)
+{
+	Grid& grid = m->grid();
+	Mesh::position_accessor_t& aaPos = m->position_accessor();
+
+	if(grid.begin<Edge>() == grid.end<Edge>())
+		return NULL;
+
+	EdgeIterator iter = grid.begin<Edge>();
+	Edge* bestEdge = *iter;
```

Comment:
maybe initialize with NULL and add a check below inside the loop:
```
if (bestEdge == NULL || dist < bestDist)
{...}
```
You could then use a for loop and avoid code duplication.